packages/core/src/pagination/pagination-button.element.ts

```diff
@@ -45,58 +45,46 @@ import styles from './pagination-button.element.scss';
  * @cssprop --box-shadow
  * @cssprop --min-height
  * @cssprop --min-width
+ * @property disabled
  */
 
-export class CdsPaginationButton extends CdsBaseButton {
+export class CdsPaginationButton extends CdsAction {
   /**
    * Sets the action from a predefined list of actions
    */
-  @property({ type: String })
-  action: 'first' | 'prev' | 'next' | 'last';
-
-  connectedCallback() {
-    super.connectedCallback();
-    this.type = 'button';
-  }
-
-  @querySlot('cds-icon', { assign: 'cds-icon-slot' }) cdsIcon: HTMLElement;
+  @property({ type: String }) action: 'first' | 'prev' | 'next' | 'last';
 
-  @query('cds-icon') private cdsIconDefault: HTMLElement;
+  @state() private direction: 'up' | 'right' | 'down' | 'left';
 
-  private get customContent() {
-    return !this.action && !this.cdsIconDefault;
+  get cdsIcon() {
+    return this.shadowRoot?.querySelector('cds-icon');
   }
 
-  render() {
-    return html`
-      <div class="private-host" cds-layout="horizontal align:center ${this.customContent ? 'p-x:sm' : ''}">
-        <slot name="cds-icon-slot">
-          ${this.action === 'next' ? html`<cds-icon shape="angle" direction="right"></cds-icon>` : ''}
-          ${this.action === 'last' ? html`<cds-icon shape="step-forward-2" direction="up"></cds-icon>` : ''}
-          ${this.action === 'prev' ? html`<cds-icon shape="angle" direction="left"></cds-icon>` : ''}
-          ${this.action === 'first' ? html`<cds-icon shape="step-forward-2" direction="down"></cds-icon>` : ''}
-        </slot>
-        <slot></slot>
-      </div>
-    `;
-  }
-
-  static get styles() {
-    return [baseStyles, styles];
-  }
+  updated(props: PropertyValues) {
+    super.updated(props);
 
-  firstUpdated(props: Map<string, any>) {
-    super.firstUpdated(props);
-    this.validateAriaLabel();
-  }
+    switch (this.action) {
```

Comment:
This seems like another good candidate for a helper function...

Reply:
Split out into helper function